octokit/search.go

```diff
@@ -0,0 +1,91 @@
+package octokit
+
+import (
+	"github.com/jingweno/go-sawyer/hypermedia"
+	"net/url"
+)
+
+var SearchURL = Hyperlink("search{/type}?q={query}{&page,per_page,sort,order}")
+
+func (c *Client) Search(url *url.URL) (searches *SearchService) {
+	searches = &SearchService{client: c, URL: url}
+	return
+}
+
+// A service to return search records
+type SearchService struct {
+	client *Client
+	URL    *url.URL
+}
+
+// Get the user search results based on SearchService#URL
+func (g *SearchService) UserSearch() (userSearchResults UserSearchResults,
+	result *Result) {
+	result = g.client.get(g.URL, &userSearchResults)
+	return
+}
+
+// Get the issue search results based on SearchService#URL
+func (g *SearchService) IssueSearch() (issueSearchResults IssueSearchResults,
+	result *Result) {
+	result = g.client.get(g.URL, &issueSearchResults)
+	return
+}
+
+// Get the repository search results based on SearchService#URL
+func (g *SearchService) RepositorySearch() (
+	repositorySearchResults RepositorySearchResults, result *Result) {
+	result = g.client.get(g.URL, &repositorySearchResults)
+	return
+}
+
+// Get the code search results based on SearchService#URL
+func (g *SearchService) CodeSearch() (
+	codeSearchResults CodeSearchResults, result *Result) {
```

Comment:
Hmm...the intention doesn't look very good. Have you tried running `go fmt ./...`? Reference: https://golang.org/cmd/gofmt/